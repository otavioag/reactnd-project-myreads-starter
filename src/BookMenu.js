import React from 'react';

const BookMenu = (props) => (
  <div className='book-shelf-changer'>
    <select value={props.shelf} onChange={(e) => {
      e.preventDefault();
      props.handleShelfChange(props.book, e.target.value);
    }}>
      <option value='move' disabled>Move to...</option>
      <option value='currentlyReading'>Currently Reading</option>
      <option value='wantToRead'>Want to Read</option>
      <option value='read'>Read</option>
      <option value='none'>None</option>
    </select>
  </div>
);

export default BookMenu;