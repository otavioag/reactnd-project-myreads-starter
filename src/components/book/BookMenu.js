import React from 'react';
import PropTypes from 'prop-types';

const BookMenu = (props) => (
  <div className='book-shelf-changer'>
    <select value={props.shelf} onChange={(e) => {
      e.preventDefault();
      props.handleShelfChange(props.book, e.target.value);
    }}>
      <option value='move' disabled>Move to...</option>
      <option value='currentlyReading'>Currently Reading</option>
      <option value='wantToRead'>Want to Read</option>
      <option value='read'>Read</option>
      <option value='none'>None</option>
    </select>
  </div>
);

BookMenu.propTypes = {
  book: PropTypes.object.isRequired,
  shelf: PropTypes.string.isRequired,
  handleShelfChange: PropTypes.func.isRequired
};

export default BookMenu;