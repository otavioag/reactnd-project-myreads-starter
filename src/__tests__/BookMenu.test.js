import React from 'react';
import { shallow } from 'enzyme';
import BookMenu from '../components/book/BookMenu';

describe('BookMenu', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<BookMenu
      book={{}}
      handleShelfChange={jest.fn()}
      shelf={'none'}/>
    );
    expect(wrapper).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('onChange works', () => {
    const handleChange = jest.fn();
    const wrapper = shallow(<BookMenu
      book={{}}
      handleShelfChange={handleChange}
      shelf={'none'}/>
    );
    wrapper.find('select').simulate('change', {target: { value : 'hello'}, preventDefault: () => (null)});
    expect(handleChange).toHaveBeenCalled();
  });
});