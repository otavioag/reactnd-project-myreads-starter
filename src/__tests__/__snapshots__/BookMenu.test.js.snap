// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookMenu renders without crashing 1`] = `
<div
  className="book-shelf-changer"
>
  <select
    aria-label="Select shelf"
    onChange={[Function]}
    value="none"
  >
    <option
      disabled={true}
      value="move"
    >
      Move to...
    </option>
    <option
      value="currentlyReading"
    >
      Currently Reading
    </option>
    <option
      value="wantToRead"
    >
      Want to Read
    </option>
    <option
      value="read"
    >
      Read
    </option>
    <option
      value="none"
    >
      None
    </option>
  </select>
</div>
`;
